household-esignet-integration-impl/src/test/java/io/mosip/esignet/household/integration/HouseHoldAuthenticatorTest.java

```diff
@@ -0,0 +1,178 @@
+package io.mosip.esignet.household.integration;
+import io.mosip.esignet.api.dto.*;
+import io.mosip.esignet.api.exception.KycExchangeException;
+import io.mosip.esignet.household.integration.dto.KycTransactionDto;
+import io.mosip.esignet.household.integration.entity.HouseholdView;
+import io.mosip.esignet.household.integration.repository.HouseholdViewRepository;
+import io.mosip.esignet.household.integration.service.HouseholdAuthenticator;
+import io.mosip.esignet.household.integration.service.HouseholdAuthenticatorHelper;
+import io.mosip.esignet.household.integration.util.ErrorConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class HouseHoldAuthenticatorTest {
+
+    @Mock
+    private HouseholdViewRepository householdViewRepository;
+
+    @Mock
+    private  HouseholdAuthenticatorHelper householdAuthenticatorHelper;
+
+    @InjectMocks
+    private HouseholdAuthenticator householdAuthenticator;
+
+
+    @Test
+    public void doKycAuth_withValidDetail_thenPass() throws Exception {
+        KycAuthDto kycAuthDto= new KycAuthDto();
+        kycAuthDto.setTransactionId("12345");
+        kycAuthDto.setIndividualId("12345");
+
+        AuthChallenge authChallenge= new AuthChallenge();
+        authChallenge.setAuthFactorType("PWD");
+        authChallenge.setFormat("alpha-numeric");
+        authChallenge.setChallenge("123456789ABC");
+        kycAuthDto.setChallengeList(Arrays.asList(authChallenge));
+
+        HouseholdView householdView= new HouseholdView();
+        householdView.setGroupName("group1");
+        householdView.setIdNumber("12345");
+        householdView.setPhoneNumber("12345");
+        householdView.setPassword(generatePasswordHash("123456789ABC"));
+
+        Mockito.when(householdViewRepository.findByIdNumber("12345")).thenReturn(Optional.of(householdView));
+
+        KycTransactionDto kycTransactionDto=new KycTransactionDto();
+        kycTransactionDto.setKycToken("12345");
+        kycTransactionDto.setPsut("12345");
+        kycTransactionDto.setHouseholdId(householdView.getHouseholdId());
+
+        Mockito.when(householdAuthenticatorHelper.setKycAuthTransaction(Mockito.anyString(),
+                Mockito.anyString(),Mockito.anyLong())).thenReturn(kycTransactionDto);
+
+        KycAuthResult kycAuthResult= householdAuthenticator.
+                        doKycAuth("12345","12345", kycAuthDto);
+        Assert.assertNotNull(kycAuthResult);
```

Comment:
Also lets assert values in the kycAuthResult

Reply:
done